src/sdk/util/request.ts

```diff
@@ -273,6 +282,29 @@ export class WalletRequest implements Serializable {
                     res.data.username = await user.getUsername();
                 }
 
+                if (req.data.kyc) {
+                    // Retrieve KYC data from user's storage if available
+                    const repository = new IdentityVerificationStorageRepository(dbConnection);
```

Comment:
instead, add a function in the `UserRequestManager` (you will need to make it a child class of `UserDataVault`) that will handle the logic of adding the `kyc` string and adding it to the data vault